app/main.py

```diff
@@ -1,13 +1,45 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [
+            {
+                "line": error["line_number"],
+                "column": error["column_number"],
+                "message": error["text"],
+                "name": error["code"],
+                "source": "flake8"
+            }
+            for error in errors
+        ],
+
+        "path": file_path,
+        "status": "passed"
+        if not any(err["filename"] == file_path for err in errors) else
```

Comment:
Your if-condition is overloaded, pass `"passed"` in case there are no errors present in the `errors` list, otherwise, pass `"failed"`. Remember that an empty list is faulty.